esp-hal/src/dma/mod.rs

```diff
@@ -256,36 +284,43 @@ impl DmaDescriptor {
         next: core::ptr::null_mut(),
     };
 
-    fn set_size(&mut self, len: usize) {
+    /// Set the size of the buffer.
+    pub fn set_size(&mut self, len: usize) {
         self.flags.set_size(len as u16)
     }
 
-    fn set_length(&mut self, len: usize) {
+    /// Set the length of the descriptor
+    pub fn set_length(&mut self, len: usize) {
         self.flags.set_length(len as u16)
     }
 
-    #[allow(unused)]
-    fn size(&self) -> usize {
+    /// Get the size of the buffer
```

Comment:
```suggestion
    /// Returns the size of the buffer.
```

Set/get is also wording that should be avoided, as those two are the two most overloaded words in the English language.

Reply:
I've update Get. You fine with Set?

Reply:
I guess I am. I'm generally not fond of docs that just repeat the function name, but I mean, what else do we need here... 😅